### PR TITLE
Sort channels in lexicographical order in packagemanifest

### PR DESCRIPTION
**Pull request by anik120**
**Motivation for the change:**
See https://github.com/operator-framework/operator-registry/issues/1069 for more details 

**Architectural changes:**
None

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage
- [ ] Sufficient end-to-end test coverage
- [ ] Bug fixes are accompanied by regression test(s)
- [ ] e2e tests and flake fixes are accompanied evidence of flake testing, e.g. executing the test 100(0) times
- [ ] tech debt/todo is accompanied by issue link(s) in comments in the surrounding code
- [ ] Tests are comprehensible, e.g. Ginkgo DSL is being used appropriately
- [ ] Docs updated or added to `/doc`
- [ ] Commit messages sensible and descriptive
- [ ] Tests marked as `[FLAKE]` are truly flaky and have an issue
- [ ] Code is properly formatted

### CODE REVIEW

**Comment by joelanford** [comment]
Nice! Should we add a test for this to avoid regressions?

File: pkg/package-server/provider/registry_test.go [hunk 1]
```diff
@@ -476,6 +477,99 @@ func TestToPackageManifest(t *testing.T) {
...
+		Name: "etcd",
+		Channels: []*api.Channel{
+			{
+				Name:    "alpha",
+				CsvName: "etcdoperator.v0.9.2",
+			},
+			{
+				Name:    "beta",
+				CsvName: "etcdoperator.v0.10.1",
+			},
```
Comment (joelanford) [inline pkg/package-server/provider/registry_test.go:491]: Does this test pass without the sort change? Seems like we may want to reverse the order of these here and check that they always end up as [alpha, beta].

Comment (joelanford) [reply pkg/package-server/provider/registry_test.go:491]: Reason being: pkg.GetChannels() returns in whatever order we define here I think:

https://github.com/operator-framework/operator-registry/blob/0a7ff74f9fc6c92be12a48a0f8e67ecfb6a8e60e/pkg/api/registry.pb.go#L175-L180

Comment (anik120) [reply pkg/package-server/provider/registry_test.go]: That's fair. I've reversed the order, the input is now [beta,alpha] with the expected output as [alpha, beta]. And with this setup, when the sort change is reversed, this test fails.

**Review by joelanford — APPROVED** [review]
/lgtm

**Pull request merged**